pytripgui/app_logic/app_callbacks.py

```diff
@@ -226,27 +226,31 @@ def one_click_callback(self):
 
         if isinstance(top_item, SimulationResultItem):
             self.app_model.viewcanvases.set_simulation_results(top_item.data, top_item.state)
-            top_item.state = self.app_model.viewcanvases.get_gui_state()
+            if top_item.state is None:
+                top_item.state = self.app_model.viewcanvases.get_gui_state()
             self.chart.set_simulation_result(top_item.data)
         elif isinstance(item, PatientItem):
             self.parent_gui.action_create_plan_set_enable(True)
             if self.app_model.viewcanvases:
                 self.app_model.viewcanvases.set_patient(top_item.data, top_item.state)
-                top_item.state = self.app_model.viewcanvases.get_gui_state()
+                if top_item.state is None:
+                    top_item.state = self.app_model.viewcanvases.get_gui_state()
         elif isinstance(item, PlanItem):
             self.parent_gui.action_create_plan_set_enable(True)
             self.parent_gui.action_create_field_set_enable(True)
             if self.app_model.viewcanvases:
                 self.app_model.viewcanvases.set_patient(top_item.data, item.state)
-                item.state = self.app_model.viewcanvases.get_gui_state()
+                if item.state is None:
+                    item.state = self.app_model.viewcanvases.get_gui_state()
         elif isinstance(item, FieldItem):
             self.parent_gui.action_create_plan_set_enable(True)
             self.parent_gui.action_create_field_set_enable(True)
             if self.is_executable(item):
                 self.parent_gui.action_execute_plan_set_enable(True)
             if self.app_model.viewcanvases:
                 self.app_model.viewcanvases.set_patient(top_item.data, item.state)
-                item.state = self.app_model.viewcanvases.get_gui_state()
+                if item.state is None:
+                    item.state = self.app_model.viewcanvases.get_gui_state()
```

Comment:
This new is statement seems a violation of DRY rule.
What about adding it like that:
```
if isinstance(item, (SimulationResultsItem, PatientItem,...)) and item.state is None:
   item.state = self.app_model.....
```

Also - a reasonable comment should be added why we change item.state in this way (i.e. explaining need of `is None` condition).

?